packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -2146,3 +2146,116 @@ export class Addr extends Op {
     stack.push(addr.publicKey);
   }
 }
+
+/* TEALv3 Ops */
+
+// immediately fail unless value top is a non-zero number
+// pops from stack: [...stack, uint64]
+export class Assert extends Op {
+  readonly line: number;
+  /**
+   * Asserts 0 arguments are passed.
+   * @param args Expected arguments: [] // none
+   * @param line line number in TEAL file
+   */
+  constructor (args: string[], line: number) {
+    super();
+    this.line = line;
+    assertLen(args.length, 0, line);
+  };
+
+  execute (stack: TEALStack): void {
+    this.assertMinStackLen(stack, 1, this.line);
+    const top = this.assertBigInt(stack.pop(), this.line);
+    if (top <= 0n) {
+      throw new RuntimeError(RUNTIME_ERRORS.TEAL.TEAL_ENCOUNTERED_ERR, { line: this.line });
+    }
+  }
+}
+
+// push immediate UINT to the stack as an integer
+// push to stack: [...stack, uint64]
+export class PushInt extends Op {
+  /**
+   * NOTE: in runtime this class is similar to Int, but from tealv3 perspective this is optimized
+   * because pushint args are not added to the intcblock during assembly processes
+   */
+  readonly uint64: bigint;
+  readonly line: number;
+
+  /**
+   * Sets uint64 variable according to arguments passed.
+   * @param args Expected arguments: [number]
+   * @param line line number in TEAL file
+   */
+  constructor (args: string[], line: number) {
+    super();
+    this.line = line;
+    assertLen(args.length, 1, line);
+    assertOnlyDigits(args[0], line);
+
+    this.checkOverflow(BigInt(args[0]), line);
+    this.uint64 = BigInt(args[0]);
+  }
```

Comment:
In Int we have support for txType map.
`const intConst = TxnOnComplete[args[0] as keyof typeof TxnOnComplete] ||
      TxnType[args[0] as keyof typeof TxnType];`
 do we need it here as well ?

Reply:
no